scipy/special/_basic.pyi

```diff
@@ -0,0 +1,271 @@
+from __future__ import annotations
+
+from typing import (
+    Callable,
+    overload,
+    Literal,
+)
+
+from ._ufuncs import psi
+from numpy import sinc as sinc
+
+from numpy.typing import NDArray
+
+from scipy._lib._util import DecimalNumber, IntNumber
+
+_ComplexNumber = DecimalNumber | complex
+
+_PointsDerivate = tuple[NDArray, NDArray]
+_ZerosOrder = tuple[NDArray, NDArray, NDArray, NDArray]
+
+__all__ = [
+    "ai_zeros",
+    "assoc_laguerre",
+    "bei_zeros",
+    "beip_zeros",
+    "ber_zeros",
+    "bernoulli",
+    "berp_zeros",
+    "bi_zeros",
+    "clpmn",
+    "comb",
+    "digamma",
+    "diric",
+    "erf_zeros",
+    "euler",
+    "factorial",
+    "factorial2",
+    "factorialk",
+    "fresnel_zeros",
+    "fresnelc_zeros",
+    "fresnels_zeros",
+    "h1vp",
+    "h2vp",
+    "ivp",
+    "jn_zeros",
+    "jnjnp_zeros",
+    "jnp_zeros",
+    "jnyn_zeros",
+    "jvp",
+    "kei_zeros",
+    "keip_zeros",
+    "kelvin_zeros",
+    "ker_zeros",
+    "kerp_zeros",
+    "kvp",
+    "lmbda",
+    "lpmn",
+    "lpn",
+    "lqmn",
+    "lqn",
+    "mathieu_even_coef",
+    "mathieu_odd_coef",
+    "obl_cv_seq",
+    "pbdn_seq",
+    "pbdv_seq",
+    "pbvv_seq",
+    "perm",
+    "polygamma",
+    "pro_cv_seq",
+    "riccati_jn",
+    "riccati_yn",
+    "sinc",
+    "stirling2",
+    "y0_zeros",
+    "y1_zeros",
+    "y1p_zeros",
+    "yn_zeros",
+    "ynp_zeros",
+    "yvp",
+    "zeta",
+]
+
+_FACTORIALK_LIMITS_64BITS: dict[int, int]
+_FACTORIALK_LIMITS_32BITS: dict[int, int]
+
+digamma = psi
+
+def diric(x: NDArray, n: DecimalNumber) -> NDArray: ...
+def jnjnp_zeros(nt: int) -> _ZerosOrder: ...
+def jnyn_zeros(n: int, nt: int) -> _ZerosOrder: ...
+def jn_zeros(n: int, nt: int) -> NDArray: ...
+def jnp_zeros(n: int, nt: int) -> NDArray: ...
+def yn_zeros(n: int, nt: int) -> NDArray: ...
+def ynp_zeros(n: int, nt: int) -> NDArray: ...
+def y0_zeros(nt: int, complex: Literal[False] = ...) -> _PointsDerivate: ...
+def y1_zeros(nt: int, complex: Literal[False] = ...) -> _PointsDerivate: ...
+def y1p_zeros(nt: int, complex: Literal[False] = ...) -> _PointsDerivate: ...
+@overload
+def jvp(v: NDArray, z: _ComplexNumber | NDArray, n: Literal[1] = ...) -> NDArray: ...
```

Comment:
This I don't understand. Why do you use a separate overload for a literal with the default value of the keyword?

Reply:
Hi, while I was reading about the type hints, I found the [PEP586](https://peps.python.org/pep-0586/) about the `Literals` and I thought that this would be the perfect case to use it, but if you think it's too much I could remove that overload.

Probably its too much, both annotations return the same result `type`.While I was reading about ..., I found ... about Bunk Beds and I thought it would be a good place to use it, but if you think it's too much I could remove that overload.

Probably its too much, both annotations return the same result `type`.

Reply:
> Probably its too much, both annotations return the same result type.

Yes indeed. If nothing changes, then an overload isn't needed and hence undesirable.

A good example of when you do need an overload is a `method=` keyword that takes a few string names. Or `order=` in numpy functions, which takes `'C', 'F', 'A', 'K'` but nothing else. Annotation that as `str` would be wrong, because the keyword does not accept an arbitrary string.